jbpm-wb-process-runtime/jbpm-wb-process-runtime-backend/src/test/java/org/jbpm/workbench/pr/backend/server/ProcessDocumentServletTest.java

```diff
@@ -0,0 +1,206 @@
+            receivedNotificationEvents.add(event);
+        }
+    };
+
+    Map<String, String> params;
+
+    @Before
+    public void setup() {
+        params = new HashMap();
+    }
+
+    @Test
+    public void testInvalidRequestParamets() throws Exception {
+        TestHttpServletRequest request = new TestHttpServletRequest(params);
+        TestHttpServletResponse response = new TestHttpServletResponse();
+
+        ProcessDocumentServlet processDocServlet = new ProcessDocumentServlet();
+        processDocServlet.setNotification(notificationEvents);
+        processDocServlet.doGet(request,
+                                response);
+
+        assertEquals(1,
+                     receivedNotificationEvents.size());
+        Object event = receivedNotificationEvents.get(0);
+        assertTrue(event instanceof NotificationEvent);
+        NotificationEvent eventReceived = (NotificationEvent) event;
+        assertEquals(MessageFormat.format(ProcessDocumentServlet.INVALID_PARAMS,
+                                          "null",
+                                          "null"),
+                     eventReceived.getNotification());
+    }
+
+    @Test
+    public void testInvalidServerTemplate() throws Exception {
+        when(kieServerIntegration.getServerClient(anyString())).thenReturn(null);
+
+        params.put("templateid",
+                   "invalidTemplateId");
+        params.put("docid",
+                   "invalidDocId");
+
+        TestHttpServletRequest request = new TestHttpServletRequest(params);
+        TestHttpServletResponse response = new TestHttpServletResponse();
+
+        ProcessDocumentServlet processDocServlet = new ProcessDocumentServlet();
+        processDocServlet.setNotification(notificationEvents);
+        processDocServlet.setKieServerIntegration(kieServerIntegration);
+        processDocServlet.doGet(request,
+                                response);
+
+        assertEquals(1,
+                     receivedNotificationEvents.size());
+        Object event = receivedNotificationEvents.get(0);
+        assertTrue(event instanceof NotificationEvent);
+        NotificationEvent eventReceived = (NotificationEvent) event;
+        assertEquals(MessageFormat.format(ProcessDocumentServlet.INVALID_SERVICES_CLIENT,
+                                          "invalidTemplateId"),
+                     eventReceived.getNotification());
+    }
+
+    @Test
+    public void testInvalidDocumentInstance() throws Exception {
+        when(kieServerIntegration.getServerClient(anyString())).thenReturn(kieServicesClient);
+        when(kieServicesClient.getServicesClient(any())).thenReturn(documentServicesClient);
+        when(documentServicesClient.getDocument(anyString())).thenReturn(null);
+
+        params.put("templateid",
+                   "kie-server-id");
+        params.put("docid",
+                   "invalidDocId");
+
+        TestHttpServletRequest request = new TestHttpServletRequest(params);
```

Comment:
I think these test classes for request/response are nice as mocking HttpServletRequest/Response is pretty tedious. IIRC spring also provides similar classes so no mocking of these two interfaces has to be done. @mswiderski wdyt? Thanks.